### PR TITLE
[Beta] - Add Redirect Middleware

### PR DESCRIPTION
**Pull request by 0xTim**
Basic Redirect Middleware for sessions - redirects users to the supplied path if they are not authenticated.

### CODE REVIEW

File: Sources/Authentication/Persist/RedirectMiddleware.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import Vapor
+
+public struct RedirectMiddleware<A>: Middleware where A: Authenticatable {
```
Comment (tanner0101) [inline Sources/Authentication/Persist/RedirectMiddleware.swift]: Class and methods should have doc comments (`///`)

File: Sources/Authentication/Persist/RedirectMiddleware.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    let path: String
+
+    public init(A authenticableType: A.Type = A.self, path: String) {
```
Comment (tanner0101) [inline Sources/Authentication/Persist/RedirectMiddleware.swift]: Static extension on Authenticatable to match the other middleware would be nice.

```swift
router.grouped(User.guard(redirectingTo: "/login")).get(...) { ... }
```

**Comment by Joannis** [comment]
@0xTim the tests aren't passing (anymore).

**Comment by 0xTim** [comment]
@Joannis yeah I'll rebase my changes back on and get the tests working and comments addressed

**Comment by 0xTim** [comment]
@Joannis @tanner0101 should be updated and tests passing again

**Comment by Joannis** [comment]
Just a small set of my notes from a slack discussion:

- `RedirectMiddleware` could be ambiguous with a Vapor/vapor middleware
- `RedirectMiddleware`'s main purpose isn't redirection but checking for authentication status
- The middleware is protecting a route against unauthorised users
- I think it could be expanded to support Authorization in the future, too
- The `login` static method could be better off without a default parameter. People have their own routes.

I'd love an overload to the `static func login` with a `Route` parameter like so:

```swift
let loginRoute = router.get("login") { request in
  return request.view().render("login.leaf")
}

let middleware = RedirectMiddleware<CustomUser>.login(at: loginRoute)
let protected = router.grouped(middleware)
```

@tanner0101 , thoughts?

File: Sources/Authentication/Persist/RedirectMiddleware.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import Vapor
+
+/// Basic middleware to redirect unauthenticated requests to the supplied path
+public struct RedirectMiddleware<A>: Middleware where A: Authenticatable {
+
+    /// The path to redirect to
+    let path: String
+
+    /**
```
Comment (tanner0101) [inline Sources/Authentication/Persist/RedirectMiddleware.swift]: `///` style, please 😁

**Pull request merged**